libraries/eventformatter/impl/src/main/kotlin/io/element/android/libraries/eventformatter/impl/StateContentFormatter.kt

```diff
@@ -0,0 +1,219 @@
+            }
+            is OtherState.RoomCreate -> {
+                if (senderIsYou) {
+                    sp.getString(R.string.state_event_room_created_by_you)
+                } else {
+                    sp.getString(R.string.state_event_room_created, senderDisplayName)
+                }
+            }
+            is OtherState.RoomEncryption -> sp.getString(io.element.android.libraries.ui.strings.R.string.common_encryption_enabled)
+            is OtherState.RoomName -> {
+                val hasRoomName = content.name != null
+                when {
+                    senderIsYou && hasRoomName -> sp.getString(R.string.state_event_room_name_changed_by_you, content.name)
+                    senderIsYou && !hasRoomName -> sp.getString(R.string.state_event_room_name_removed_by_you)
+                    !senderIsYou && hasRoomName -> sp.getString(R.string.state_event_room_name_changed, senderDisplayName, content.name)
+                    else -> sp.getString(R.string.state_event_room_name_removed, senderDisplayName)
+                }
+            }
+            is OtherState.RoomThirdPartyInvite -> {
+                if (content.displayName == null) {
+                    Timber.e("RoomThirdPartyInvite undisplayable due to missing name")
+                    return null
+                }
+                if (senderIsYou) {
+                    sp.getString(R.string.state_event_room_third_party_invite_by_you, content.displayName)
+                } else {
+                    sp.getString(R.string.state_event_room_third_party_invite, senderDisplayName, content.displayName)
+                }
+            }
+            is OtherState.RoomTopic -> {
+                val hasRoomTopic = content.topic != null
+                when {
+                    senderIsYou && hasRoomTopic -> sp.getString(R.string.state_event_room_topic_changed_by_you, content.topic)
+                    senderIsYou && !hasRoomTopic -> sp.getString(R.string.state_event_room_topic_removed_by_you)
+                    !senderIsYou && hasRoomTopic -> sp.getString(R.string.state_event_room_topic_changed, senderDisplayName, content.topic)
+                    else -> sp.getString(R.string.state_event_room_topic_removed, senderDisplayName)
+                }
+            }
+            is OtherState.Custom -> when (renderingMode) {
+                RenderingMode.RoomList -> {
+                    Timber.v("Filtering timeline item for room state change: $content")
+                    null
+                }
+                RenderingMode.Timeline -> {
+                    "Custom event ${content.eventType}"
+                }
+            }
+            OtherState.PolicyRuleRoom -> when (renderingMode) {
+                RenderingMode.RoomList -> {
+                    Timber.v("Filtering timeline item for room state change: $content")
+                    null
+                }
+                RenderingMode.Timeline -> {
+                    "PolicyRuleRoom"
```

Comment:
Are we sure we want to add these? They look quite weird in the timeline. 
<img width="507" alt="image" src="https://github.com/vector-im/element-x-android/assets/480955/21966abc-9b4c-4f7e-9a38-2d15b4be5c62">

Reply:
Yes, for now the application is displaying what the SDK is providing, i.e. not a lot.
I prefer to revisit this later.